From 0cfd77f103e530fccfcedd4f4754060c6d94924f Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 14:38:15 -0400
Subject: [ethernet] enic: slight optimization of addr compare

Message-id: <1409668716-4814-4-git-send-email-sassmann@redhat.com>
Patchwork-id: 90630
O-Subject: [RHEL7.1 PATCH 03/24] net: enic: slight optimization of addr compare
Bugzilla: 1107934
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: dingtianhong <dingtianhong@huawei.com>

Use possibly more efficient ether_addr_equal
to instead of memcmp.

Cc: Christian Benvenuti <benve@cisco.com>
Cc: Sujith Sankar <ssujith@cisco.com>
Cc: Govindarajulu Varadarajan <govindarajulu90@gmail.com>
Cc: Neel Patel <neepatel@cisco.com>
Cc: Nishank Trivedi <nistrive@cisco.com>
Signed-off-by: Ding Tianhong <dingtianhong@huawei.com>
Acked-by: Govindarajulu Varadarajan <govindarajulu90@gmail.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 9130ac61a5a909f6e463a18abafa99d4a2b5842f)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/cisco/enic/enic_pp.c b/drivers/net/ethernet/cisco/enic/enic_pp.c
index 43464f0..e6a8319 100644
--- a/drivers/net/ethernet/cisco/enic/enic_pp.c
+++ b/drivers/net/ethernet/cisco/enic/enic_pp.c
@@ -162,7 +162,7 @@ static int enic_are_pp_different(struct enic_port_profile *pp1,
  return strcmp(pp1->name, pp2->name) | !!memcmp(pp1->instance_uuid,
   pp2->instance_uuid, PORT_UUID_MAX) |
   !!memcmp(pp1->host_uuid, pp2->host_uuid, PORT_UUID_MAX) |
-  !!memcmp(pp1->mac_addr, pp2->mac_addr, ETH_ALEN);
+  !ether_addr_equal(pp1->mac_addr, pp2->mac_addr);
 }
 
 static int enic_pp_preassociate(struct enic *enic, int vf,
-- 
1.7.1